packages/concerto-core/lib/decoratormanager.js

```diff
@@ -253,6 +253,178 @@ class DecoratorManager {
         newModelManager.fromAst(decoratedAst);
         return newModelManager;
     }
+
+    /**
+     * Applies all the decorator commands from the DecoratorCommandSet
+     * to the ModelManager.
+     * @param {*} decoratorCommandSet the DecoratorCommandSet object
+     * @returns {Object} a new model manager with the decorations applied
+     */
+    static getDecoratorMaps(decoratorCommandSet) {
+        const namespaceCommandsMap = new Map();
+        const declarationCommandsMap = new Map();
+        const propertyCommandsMap = new Map();
+        const mapElementCommandsMap = new Map();
+        const typeCommandsMap = new Map();
+
+        decoratorCommandSet.commands.map(decoratorCommand => {
+            if(decoratorCommand.target.type) {
+                if (typeCommandsMap.has(decoratorCommand.target.type)) {
+                    typeCommandsMap.get(decoratorCommand.target.type).push(decoratorCommand);
+                } else {
+                    typeCommandsMap.set(decoratorCommand.target.type, [{...decoratorCommand}]);
+                }
```

Comment:
You can optimize this a little by removing one of the lookups ...
```suggestion
                const typeCommands = typeCommandsMap.get(decoratorCommand.target.type);
                if (typeCommands) {
                    typeCommands.push(decoratorCommand);
                } else {
                    typeCommandsMap.set(decoratorCommand.target.type, [{...decoratorCommand}]);
                }
              }
```